tools/code_format/glint.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+ERRORS=
+MISSING_NEWLINE=0
+MIXED_TABS_AND_SPACES=0
+TRAILING_WHITESPACE=0
+
+
+check_mixed_tabs_spaces () {
+    local spaced tabbed
+    tabbed=$(grep -ce '^\t' "$1")
+    spaced=$(grep -ce '^ ' "$1")
+    if [[ $tabbed -gt 0 ]] && [[ $spaced -gt 0 ]]; then
+	echo "mixed tabs and spaces: ${1}"
+	ERRORS=yes
+	((MIXED_TABS_AND_SPACES=MIXED_TABS_AND_SPACES+1))
+    fi
+}
+
+check_new_line () {
+    test "$(tail -c 1 "$1" | wc -l)" -eq 0 && {
```

Comment:
and this - its using a kinda bash trick to make sure that there is a new line - regardless of what type of new line it is